Flaky test: test_refreshable_mv/test.py::test_refreshable_mv_in_replicated_db

_Important: This issue was automatically generated and is used by CI for matching failures. DO NOT modify the body content. DO NOT remove labels._

Test name: test_refreshable_mv/test.py::test_refreshable_mv_in_replicated_db
Failure reason: DB::Exception: Refreshable view re.append (b04cb54b-0754-4203-9f99-57310d4f4dbb) doesn't exist
CI report: [Integration tests (amd_asan, db disk, old analyzer, 5/6)](https://s3.amazonaws.com/clickhouse-test-reports/json.html?PR=92372&sha=b05cc66744836302d69f3929cc8f3b953d8f3089&name_0=PR&name_1=Integration%20tests%20%28amd_asan%2C%20db%20disk%2C%20old%20analyzer%2C%205%2F6%29)

CIDB statistics: [cidb](https://play.clickhouse.com/play?user=play&run=1#V0lUSAogICAgOTAgQVMgaW50ZXJ2YWxfZGF5cwpTRUxFQ1QKICAgIHRvU3RhcnRPZkRheShjaGVja19zdGFydF90aW1lKSBBUyBkYXksCiAgICBjb3VudCgpIEFTIGZhaWx1cmVzLAogICAgZ3JvdXBVbmlxQXJyYXkocHVsbF9yZXF1ZXN0X251bWJlcikgQVMgcHJzLAogICAgYW55KHJlcG9ydF91cmwpIEFTIHJlcG9ydF91cmwKRlJPTSBjaGVja3MKV0hFUkUgKG5vdygpIC0gdG9JbnRlcnZhbERheShpbnRlcnZhbF9kYXlzKSkgPD0gY2hlY2tfc3RhcnRfdGltZQogICAgQU5EIHRlc3RfbmFtZSA9ICd0ZXN0X3JlZnJlc2hhYmxlX212L3Rlc3QucHk6OnRlc3RfcmVmcmVzaGFibGVfbXZfaW5fcmVwbGljYXRlZF9kYicKICAgIC0tIEFORCBjaGVja19uYW1lID0gJ0ludGVncmF0aW9uIHRlc3RzIChhbWRfYXNhbiwgZGIgZGlzaywgb2xkIGFuYWx5emVyLCA1LzYpJwogICAgQU5EIHRlc3Rfc3RhdHVzIElOICgnRkFJTCcsICdFUlJPUicpCiAgICBBTkQgKChoZWFkX3JlZiA9ICdtYXN0ZXInIEFORCBwdWxsX3JlcXVlc3RfbnVtYmVyID0gMCkgT1IgcHVsbF9yZXF1ZXN0X251bWJlciAhPSAwKQpHUk9VUCBCWSBkYXkKT1JERVIgQlkgZGF5IERFU0MK)

Test output:
```
File: test_refreshable_mv/test.py:104 - in test_refreshable_mv_in_replicated_db
    node.query(
File: helpers/cluster.py:4441 - in query
    return self.client.query(
File: helpers/client.py:40 - in wrap
    return func(self, *args, **kwargs)
File: helpers/client.py:80 - in query
    ).get_answer()
File: helpers/client.py:269 - in get_answer
    raise QueryRuntimeException(
E   helpers.client.QueryRuntimeException: Client failed! Return code: 36, stderr: Received exception from server (version 25.12.1):
E   Code: 36. DB::Exception: Received from 172.16.6.10:9000. DB::Exception: Refreshable view re.append (b04cb54b-0754-4203-9f99-57310d4f4dbb) doesn't exist. Stack trace:
E   
E   0. ./contrib/llvm-project/libcxx/include/__exception/exception.h:113: Poco::Exception::Exception(String const&, int) @ 0x000000004d8dc831
E   1. ./ci/tmp/build/./src/Common/Exception.cpp:131: DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0x0000000024748cb3
E   2. DB::Exception::Exception(String&&, int, String, bool) @ 0x000000000e6e8137
E   3. DB::Exception::Exception(PreformattedMessage&&, int) @ 0x000000000e6e7267
E   4. DB::Exception::Exception<String>(int, FormatStringHelperImpl<std::type_identity<String>::type>, String&&) @ 0x000000000e6f0278
E   5. ./ci/tmp/build/./src/Interpreters/InterpreterSystemQuery.cpp:1897: DB::InterpreterSystemQuery::getRefreshTasks() @ 0x0000000035a0dab7
E   6. ./ci/tmp/build/./src/Interpreters/InterpreterSystemQuery.cpp:815: DB::InterpreterSystemQuery::execute() @ 0x00000000359fcf07
E   7. ./ci/tmp/build/./src/Interpreters/executeQuery.cpp:1738: DB::executeQueryImpl(char const*, char const*, std::shared_ptr<DB::Context>, DB::QueryFlags, DB::QueryProcessingStage::Enum, std::unique_ptr<DB::ReadBuffer, std::default_delete<DB::ReadBuffer>>&, std::shared_ptr<DB::IAST>&, std::shared_ptr<DB::ImplicitTransactionControlExecutor>, std::function<void ()>, DB::QueryResultDetails&) @ 0x00000000359889ee
E   8. ./ci/tmp/build/./src/Interpreters/executeQuery.cpp:1959: DB::executeQuery(String const&, std::shared_ptr<DB::Context>, DB::QueryFlags, DB::QueryProcessingStage::Enum) @ 0x000000003597f0c3
E   9. ./ci/tmp/build/./src/Server/TCPHandler.cpp:776: DB::TCPHandler::runImpl() @ 0x000000003efef2cd
E   10. ./ci/tmp/build/./src/Server/TCPHandler.cpp:2866: DB::TCPHandler::run() @ 0x000000003f032656
E   11. ./ci/tmp/build/./base/poco/Net/src/TCPServerConnection.cpp:40: Poco::Net::TCPServerConnection::start() @ 0x000000004dad71af
E   12. ./ci/tmp/build/./base/poco/Net/src/TCPServerDispatcher.cpp:115: Poco::Net::TCPServerDispatcher::run() @ 0x000000004dad7f17
E   13. ./ci/tmp/build/./base/poco/Foundation/src/ThreadPool.cpp:205: Poco::PooledThread::run() @ 0x000000004d9ccaeb
E   14. ./base/poco/Foundation/src/Thread_POSIX.cpp:341: Poco::ThreadImpl::runnableEntry(void*) @ 0x000000004d9c63c8
E   15. asan_thread_start(void*) @ 0x000000000e68e1a7
E   16. ? @ 0x0000000000094ac3
E   17. ? @ 0x00000000001268c0
E   . (BAD_ARGUMENTS)
E   (query: system test view re.append set fake time '2040-01-01 00:00:01')
```

--- a/src/Interpreters/InterpreterDropQuery.cpp
+++ b/src/Interpreters/InterpreterDropQuery.cpp
@@ -569,8 +569,15 @@ BlockIO InterpreterDropQuery::executeToDatabaseImpl(const ASTDropQuery & query, 
                     /// In Replicated databases, if we drop an MV first, its dropInnerTableIfAny() tries to
                     /// drop the inner table via executeDropQuery which fails because the replicated DDL path
                     /// rejects secondary queries. Dropping inner tables first makes dropInnerTableIfAny() a no-op.
-                    bool a_is_inner = a.first.table_name.starts_with(".inner_id.") || a.first.table_name.starts_with(".inner.");
-                    bool b_is_inner = b.first.table_name.starts_with(".inner_id.") || b.first.table_name.starts_with(".inner.");
+                    /// Note: refreshable MVs also create `.tmp.inner_id.*` temporary tables during refresh,
+                    /// and `dropInnerTableIfAny` drops those too, so they must also be classified as inner.
+                    auto is_inner_table_name = [](const String & name)
+                    {
+                        return name.starts_with(".inner_id.") || name.starts_with(".inner.")
+                            || name.starts_with(".tmp.inner_id.") || name.starts_with(".tmp.inner.");
+                    };
+                    bool a_is_inner = is_inner_table_name(a.first.table_name);
+                    bool b_is_inner = is_inner_table_name(b.first.table_name);
                     if (a_is_inner != b_is_inner)
                         return a_is_inner;
 

